usr/share/rear/layout/prepare/default/300_map_disks.sh

```diff
@@ -309,17 +309,23 @@ while true ; do
     esac
 done
 
-# Mark unmapped 'disk' devices and 'multipath' devices in the LAYOUT_FILE as done
-# so that unmapped 'disk' and 'multipath' devices will not be recreated
+# Mark unmapped 'disk', 'multipath' and 'opaldisk' devices in the
+# LAYOUT_FILE as done so that such devices will not be recreated
 # except we are no longer in migration mode when the user confirmed an
 # identical disk mapping and proceeding without manual configuration:
 if is_true "$MIGRATION_MODE" ; then
     while read keyword device junk ; do
         is_mapping_source "$device" && continue
-        LogUserOutput "Disk $device and all dependant devices will not be recreated"
-        mark_as_done "$device"
-        mark_tree_as_done "$device"
-    done < <( grep -E "^disk |^multipath " "$LAYOUT_FILE" )
+        if [[ "$keyword" == "opaldisk" ]]; then
+            Log "TCG Opal 2-compliant self-encrypting disk $device will not be recreated"
+            # Note: dependent devices might still be recreated if $device is mapped as a (non-encrypting) disk
+            mark_as_done "opaldisk:$device"
+        else
```

Comment:
Only a personal coding style side note:
In loops I prefer to `continue` with the next loop iteration
instead of `if ... then ... else ... fi` clauses like
```
while ... ; do
    if SPECIAL_CASE. ; then
        ...
        continue
    fi
    ...
done
```
cf. the basic idea behind "Return early, return often"
in https://github.com/rear/rear/wiki/Coding-Style
but that is a matter of personal coding style taste.

Reply:
I'm OK with early returns to conditionally skip entire rear scripts. I'm also fine using goto-style for exception handling in languages without support for try/catch constructs.

Otherwise, I'd avoid using goto-like statements. Short-circuiting loops might easily be overlooked when adding code at the bottom of the loop later on.

Reply:
As I said: It is a matter of personal coding style taste.
I think @schlomo agrees with your pint of view that
```
short-circuiting loops might easily be overlooked
when adding code at the bottom of the loop later on
```
while my point of view contradicts because I think
```
whoever adds some code at the bottom of
"a closed piece of code" (e.g. a loop, an if clause, a function, ...)
cannot expect that his addon works as intended
unless he had examined what all the code above
his addon actually does
```
and therefore I prefer readability over all
because I must read and understand all the code
before I can do any changes.